crates/nu-parser/src/parser.rs

```diff
@@ -2890,6 +2890,10 @@ pub fn parse_import_pattern(working_set: &mut StateWorkingSet, spans: &[Span]) -
     }
 }
 
+// Parse `spans[spans_idx..]` into a variable, with optional type annotation.
+// If the name of the variable ends with a colon (no space in-between allowed), then a type annotation
+// can appear after the variable, in which case the colon is stripped from the name of the variable.
+// `spans_idx` is updated to point to the last span that has been parsed.
```

Comment:
May be we can make them starts with `///`(they can be function docstring)

Reply:
Good catch. There're some code that still uses `//` for top-level comments, but we can deal with that later.